wireguard-manager.sh

```diff
@@ -1100,8 +1100,15 @@ else
       if [ -z "${NEW_CLIENT_NAME}" ]; then
         NEW_CLIENT_NAME="$(openssl rand -hex 50)"
       fi
-      LASTIPV4=$(grep "/32" ${WIREGUARD_CONFIG} | tail -n1 | awk '{print $3}' | cut -d "/" -f 1 | cut -d "." -f 4)
-      LASTIPV6=$(grep "/128" ${WIREGUARD_CONFIG} | tail -n1 | awk '{print $3}' | cut -d ":" -f 5 | cut -d "/" -f 1)
+      # Look for an unused IP address.
+      FIND_UNUSED_IP=$(grep -E -o "^AllowedIPs = $(head -n1 ${WIREGUARD_CONFIG} | awk '{print $2}' | cut -d\. -f1,2,3)\.[0-9]{1,3}" ${WIREGUARD_CONFIG} | awk '{print $3}' | cut -d\. -f4 | sort -n | awk '{for(i=p+1; i<$1; i++) print i} {p=$1}' | grep -E -v '^1$')
```

Comment:
Everything else is fine, can you please improve this line.

Reply:
The simpler the code, the better, and while this appears to be a touch complex, it does the job, but I'm sure we can make it much simpler.